src/utils/fundamentals/fundamentals.js

```diff
@@ -0,0 +1,199 @@
+import * as utils from '../network/utils'
+import { searchData } from '../network'
+
+/**
+ * Make request and search data
+ * @function countries
+ * @param {object} params - object of parameters:
+ * @param {string} params.countryCode 
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function countries({ code }, callback) {
+  const params = {
+    AMaaSClass: 'fundamentalCountries',
+    query: {countryCode: [code]}
+  }
+  //if code is not specified, retun all countries 
+  let promise = searchData(params).then(result => {
+    if (typeof callback === 'function') {
+      callback(null, result)
+    }
+    return result
+   }).catch((err) => {
+     console.log(err.message);
+  });
+  if (typeof callback !== 'function') {
+    // return promise if callback is not provided
+    return promise
+  }
+  promise.catch(error => callback(error))
+}
+
+/**
+ * Make request and search data
+ * @function calcBusinessDate
+ * @param {object} params - object of parameters:
+ * @param {string} params.codes - A country code string or an array of country codes strings 
+ * @param {string} params.date - start date
+ * @param {string} params.offset
+ * @param {string} params.invalidDates - an invalid date string or an array of invalid dates strings 
+ * @param {function} [callback] - Called with two arguments (error, result) on completion. `result` is a country name returned or country names. Omit to return Promise
+ * @returns {Promise|null} 
+ */
+export function calcBusinessDate({date, codes, offset, invalidDates}, callback) //invalid dates
+{
+    if(!date){
+        return "Must specify start_date in querystring"
+    }else if(!codes){
+        return "Must specify at least one country_code in querystring"
+    }else if(!offset){
+        return "Must specify offset in querystring"
+    }else if(!Number.isInteger(offset)) {
+         return "Offset must be an integer"
+    }
+
+    let code=""
+    if(Array.isArray(codes)){
+      for(var i = 0; i < codes.length;i++) {
+            code+=codes[i]+","
+       } 
+     code = code.substring(0, code.length - 1);
+   }else{
+    code=codes
+   }
+   
+   let invalidDate=""
+   if(invalidDates){
+    if(Array.isArray(invalidDates)){
+      for(var i = 0; i < invalidDates.length;i++) {
+            invalidDate+=invalidDates[i]+","
+       } 
+     invalidDate = invalidDate.substring(0,invalidDate.length - 1);
+    }else{
+      invalidDate=invalidDates
+    }
+   }
+    
+   let params 
+   if(!invalidDates){
```

Comment:
You can make this cleaner:

```javascript
const params = {
  AMaaSClass: 'fundamentalBusinessDate',
  query: { startDate: [date], countryCode: codes, offset: [offset], invalidDate: invalidDate ? [invalidDate] : []  }
}
// The invalidDate part means that if the user has supplied invalidDate, then put it in an array otherwise use an empty array.
```